clickhouse/client.cpp

```diff
@@ -81,9 +81,36 @@ std::ostream& operator<<(std::ostream& os, const ClientOptions& opt) {
     return os;
 }
 
+
+namespace {
+
+std::unique_ptr<SocketFactory> GetSocketFactory(const ClientOptions& opts) {
+    (void)opts;
+#if defined(WITH_OPENSSL)
+    if (opts.ssl_options.use_ssl)
+        return std::make_unique<SSLSocketFactory>();
+    else
+#endif
+        return std::make_unique<NonSecureSocketFactory>();
+}
+
+std::unique_ptr<SleepImpl> GetDefaultSleepImpl() {
```

Comment:
Please consider changing the interface to get rid of the `SleepImpl`
https://github.com/ClickHouse/clickhouse-cpp/pull/147/files#r799244776